lineapy/plugins/taskgen.py

```diff
@@ -120,19 +120,26 @@ def get_session_task_definition(
             for var in session_input_variables
         ]
 
+        return_vars = [
+            nc.safename
+            for nc in session_artifacts.usercode_nodecollections
+            if isinstance(nc, ArtifactNodeCollection)
+        ]
+
         raw_function_call_block = (
             BaseSessionWriter().get_session_function_callblock(
                 session_artifacts
             )
         )
+        # Call module's run session function and unpack the artifacts from it
         function_call_block = (
-            f"artifacts = {pipeline_name}_module.{raw_function_call_block}"
+            f"artifacts = {pipeline_name}_module.{raw_function_call_block}\n"
+            + "\n".join(
+                f'{nc.safename} = artifacts["{nc.name}"]'
+                for nc in session_artifacts.usercode_nodecollections
+                if isinstance(nc, ArtifactNodeCollection)
+            )
```

Comment:
I see the logic is correct here to extract variables from the dictionary so you can put it into `task_local_pickle_ser.jinja` . 

However, the `function_call_block` now does more things than it looks like; maybe use a new variable name or put unpack logic into different variable (`unpack_block`?) and rename `call_block` in `TaskDefinition` as `execute_block = function_call_block + unpack_block` .

Reply:
yeah this is a nice idea, I'll probably switch to this

Reply:
But i think i will edit task_function.jinja instead and make it more generalizable by calling it a "pre_call_block" and "post_call_block".